### ISSUE TITLE
Support the ability to change build context

### ISSUE THREAD

**Issue by bkendzior** [earthly/earthly#1221]
I'm curious if there's a way to change the build context when running earthly.

i.e. my repository is organized as such:
```
/repo_root
|- ./docker
 |- Earthfile
 |- configurations, scripting, etc
|- ./src
 |- code I want to COPY into my container
```

docker build allows for you to specify a build context of `..` for this particular use-case, curious if Earthly supports anything similar or if the build context is purely an artifact of the placement of the Earthfile

**Comment by alexcb** [comment]
as a work around, the following might work until we have a better solution:

```
run:
    FROM alpine
    COPY --dir +hack/src .
    RUN cat src/code

hack:
    LOCALLY
    SAVE ARTIFACT ../src src
```

**Comment by vladaionescu** [comment]
We might not want to support this as a matter of convention. We want the context of the build to always be the current directory to make it easy for the user to follow the logic.

**Comment by bjaanes** [comment]
Or perhaps make it even more explicit instead with `--context`, `-c` set to the path where you want the context to be run from.
That way it is still possible to run the Earthfile from anywhere with the `earthly ./path/to/earthfile/dir+target` (and not having two ways of specificing where the Earthfile is) and it would have a slightly similar way of doing it to Docker (only the context is not required and is by default the location of the Earthfile).

**Comment by andrewsykim** [comment]
After mulling over this for a few hours, I think we need to give this more thought. Being able to set the local context can be useful for many cases as we already mentioend but it can negatively impact the portability and simplicity of Earthfiles. I moved the issue back to "Under Consideration" as we think about the right way to approach this.

**Comment by adamgordonbell** [comment]
Rather than copy from a parent, it would be useful to have the source control root as a var in Earthfiles. Then in a mono repo setup, you could refer to other source dependencies, but still, you wouldn't harm reproducibility because you are never outside of your source control.
```
/repo_root
|- ./docker
 |- Earthfile
 |- configurations, scripting, etc
|- ./src
 |- code
```

Then you could do something like :  
```
COPY $ROOT/src/code .
```

I hit this issue with building this[ go project](https://github.com/adamgordonbell/cloudservices/tree/main/activity-client) but I suspect having a way to reference paths from the root would be useful in many mono-repo setup where their are dependencies between sibling paths.

**Comment by brandonSc** [comment]
Wonder if both options should be supported. 

`../my-other-dir` is pretty intuitive and I think a lot of users may try that and expect that it might work. 

Something like `$ROOT` would be good for more deeply nested scenarios, e.g. `../../../my-other-thing` could get confusing.

Note: support for some or all of these ideas may be especially useful in monorepos.

**Comment by vladaionescu** [comment]
I agree - both should be supported. I wonder if there should be some safety thing to not let users escape the directory tracked by git, unless they pass some flag to `COPY`.

**Comment by ryuheechul** [comment]
a workaround that doesn't involve with `LOCALLY` hack but requires putting an extra Earthfile at the parent directory

```Earthfile
# at a sub directory, say `./sub/Earthfile`

list:
  FROM alpine
  COPY --dir ../+project-root/code ./
  RUN ls code
```

```Earthfile
# at the parent directory, say `./sub/../Earthfile`
# (yeah basically, `./Earthfile`, you get the point) 

project-root: # name this whatever you want
  FROM scratch
  COPY . /code # so as this directory
  SAVE ARTIFACT /code code # and so as this artifact name
```
